Documentation/contributors/design-proposals/operator-config.md

```diff
@@ -0,0 +1,338 @@
+ 
+ ```yaml
+   config:
+   - selector:                     
+      matchLabels:
+        k8s-app: prometheus-operator
+ ```
+
+ But this `Subscription` config with the selector **will not match**:
+ 
+ ```yaml
+   config:
+   - selector:                     
+      matchLabels:
+        olm.cahash: 123
+ ```
+ 
+Because matching is determined from the pod template and not from the real pod on the cluster. Similarly, the configuration will only apply to pods defined by ClusterServiceVersion that has been installed from the Subscription, and will not apply to any other pods, even if the the pod has a matching label.
+ 
+**Upgrades:** Operator configuration is persisted between updates to the operator, but only if the updated operator's pods continue to match the defined selector. Operator authors should not remove previously-defined template labels unless they wish to prevent previously-defined config from applying.
+
+#### Scenario: Enable debug logs for an operator
+
+In this scenario, we have an operator that has been written to have a command flag set to enable debug logs: `-v=4`.
+
+The relevent section of the `ClusterServiceVersion` (note the new `$(ARGS)`)
+
+```yaml
+kind: ClusterServiceVersion
+spec:
+  install:
+    spec:
+      deployments:
+      - name: prometheus-operator
+        spec:
+          template:
+            metadata:
+              labels:
+                k8s-app: prometheus-operator
+            spec:
+              containers:
+              - name: prometheus-operator
+                args:
+                - -namespaces=$(NAMESPACES)
+                - $(ARGS)
+                env:
+                - name: NAMESPACES
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.annotations['olm.targetNamespaces']
+```
+This can then be configured, optionally, from the subscription:
+
+```yaml
+kind: Subscription
+metadata:
+  name: prometheus
+spec:
+  pacakge: prometheus
+  channel: alpha
+  config:
+  - selector:                     
+      matchLabels:
+        k8s-app: prometheus-operator
+    env:
+    - name: ARGS
+      value: "-v=4"
+```
+
+The `Deployment` object will then be updated by OLM:
+
+```
+kind: Deployment
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: prometheus-operator
+    spec:
+      containers:
+      - name: prometheus-operator
+        args:
+        - -namespaces=$(NAMESPACES)
+        - $(ARGS)
+        env:
+        - name: NAMESPACES
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.annotations['olm.targetNamespaces']
+        - name: ARGS
+          value: "-v=4"
+```
+
+When the operator updates to a newer version, it will still be configured with `-v=4	` (though it's up to the operator author whether that continues to have the desired effect).
+
+#### Openshift Notes
+
+When running in Openshift, OLM will fill in the config for env vars:
+
+- `HTTP_PROXY`
+- `HTTPS_PROXY`
+- `NO_PROXY`
```

Comment:
@knrc somewhat relevant to your socks proxy question for istio.....there's still no socks proxy field being added to the global proxy config today that i know of, but if one is added, this is how i'd expect the value to get passed down to your OLM operator.